### DISCUSSION TITLE
Trouble with iOS Devices CalDav/CardDav on one newly installed AIO-Instance

### DISCUSSION THREAD

**Discussion by neocult-de** [nextcloud/all-in-one#5210] [Questions] [answered]
Hi everyone,

I have several Nextcloud-AIO instances, behind npm and/or with dedicated ip. On my latest instance I can´t get iOS to connect the Nextcloud profiles (CardDav+CalDav) to work anymore. Error: **The operation couldn´t be completed: DAAccountValidationDomain error 102**

On the other instances it works. All are on latest stable. I put the current instance on a dedicated IP and even configured the PTR / Reverse Entry for the Domain. I have LetsEncrypt certificates auto-generated as usual, CalDav/CardDav URLs work normally if called in a Browser, Thunderbird, Android, but just iOS doesn´t seem to work, and only on this one instance.

Anyone experienced that with Nextcloud-AIO? How would I debug this further? Checking the forwarding of URLs, use the iOS-specific Addresses / use the Profile under Mobile & Desktop both don´t work with the given error message.

As it is set by default overwriteprotocol https is set in the config, which sometimes can be an issue. And it is not 2FA, i tried directly with more than one user account and direct password and app password either way.

Looking forward to any hint. Thank you very much.

**Comment by neocult-de** [comment]
I was able to narrow down the issue to getting that issue only if it is the root-Domain. All other instances on a subdomain have no issue and if I change the current domain to a subdomain of it, it starts to work as intended under iOS.

Still only iOS seems to check something more about a domain than just using the settings provided by the profile as any other app. But then it might not be Nextcloud-AIO related, but something about service discovery and other systems configured within the domain.

**Comment by neocult-de** [comment, marked as answer]
I finally figured it out over the weekend, after waiting for the DNS Caching to be a non-issue. If you are with a hoster, where Domains might already have SRV and TXT-Records for caldav and carddav services and they don´t match your nextcloud instance, an iOS Device will refuse working and throw error "The operation couldn´t be completed: DAAccountValidationDomain error 102" when trying to add a Nextcloud profile.

If you want to set proper discovery entries in DNS you have to set SRV AND TXT entries for the domains:
**_caldav._tcp.yourdomain.com.
_caldavs._tcp.yourdomain.com.
_carddav._tcp.yourdomain.com.
_carddavs._tcp.yourdomain.com.**

Whereby the SRV records have
**Priority: 0, Weight: 0, Port 443, Target:  yourdomain.com**
and the TXT records have
**path=/remote.php/dav/**


For future reference I wanted to have that out there, as using search engines it is hard to find anything related to that error code and other platforms don´t block using nextcloud with wrong or no dns entries at all and there is no direct link to this if searching in general for iOS and Nextcloud Sync issues.

**Reply by artwaw** [reply]
Thank you for that, I tried to import mobile profile from the next cloud to apple device and got hit by that error!

**Reply by papalamg** [reply]
Hallo, i have also an error DAAccountValidationDomain error 100.
In my local network I have a lets encrypt wildcard certificate which is shown as valid for my next.damoain.com when I log in via web interface.
I have a local DNS (adguard) where I have a Local DNS Records [A/AAAA] from next.mydomain.com on the ip address of my server.
How to set the SRV and TXT entries? Must this be done in adguard? How?

**Reply by neocult-de** [reply]
@papalamg The txt and srv entries should be done in the official dns records for your domain (so not adguard). They hint at the port and your next.damoain.com 

You can then still rewrite your next.damoain.com in your Adguard as you already have it so that the domain is resolved to your local ip of the nextcloud (I have the same setup).

**Comment by papalamg** [comment]
Hallo neocult,
I tried your answer. Before I got the error DAAccountValidationDomain:100, after the setting of the records I get this error: DAAccountValidationDomain:101 and it takes some more time until this error comes.
I am not sure if I made the right settings at my provider:
my nextcloud is on a truenas machine as an app (docker-based) and I can reach it only at home under: https://next.lang-w.de:30027
I installed a valid lets encrypt certificate:
![image](https://github.com/user-attachments/assets/7650b22e-1bb1-48f1-b616-fde28a340fd3)

this is the setting of the txt-records:
![image](https://github.com/user-attachments/assets/8a084151-d549-4c08-bbba-468a0ce3aee8)
and this of the SRV-records:
![image](https://github.com/user-attachments/assets/9f0f78e5-4269-48d7-9053-0150e225dba9)
and this are all the settings:
![image](https://github.com/user-attachments/assets/ad591954-5148-483a-be8f-f92cf805ecc6)

I am not sure if the settings for the DNS is OK?

here the check for https://next.lang-w.de:30027/well-known/carddav
![image](https://github.com/user-attachments/assets/843dbd66-1bb6-45d5-9117-55044e0f3c3d)

**Reply by neocult-de** [reply]
Your service entries have to be in the root of the domain. E.g. _caldav._tcp (no next) and then the domain for the SRV entries must be next.lang-w.de where your Nextcloud is actually reachable.

Then in your local network the profile should be fine. Outside your network as the Cloud is not reachable, verification is of course not possible.

**Reply by papalamg** [reply]
I tried now this for all 4 entries (_caldav, _caldavs, _carddav, _carddavs) without success :-(
DAAccountValidationDomain:101
![image](https://github.com/user-attachments/assets/b8d1b9c4-e5c0-4eed-ba5e-6a620885c7b4)
the text entries are with the full domain name. Correct?
Can I verify if the error is in the TXT or SRV entry?

**Comment by gloewl** [comment]
I solved it for me.
I had the same issue and tried everything I could find in this thread and all the others. It turned out that the 2FA that I enabled for my user in Nextcloud was the "problem". The profile could not be installed as the 2FA-mechanism was not triggered by macOS. I disabled 2FA in Nextcloud, and boom: It worked.

**Reply by papalamg** [reply]
I don't have 2FA in nextcloud. Have you set the TXT and SRV records?

**Reply by gloewl** [reply]
Yes, I set them as described. But I just migrated my Nextcloud instance from another server where everything worked like a charm even without those DNS settings. I guess that I could reset them and it would still work, I simple enjoy a working system now and don’t want to experiment with it, that’s why I leave it as it is for now.

**Comment by papalamg** [comment]
i am still trying since weeks to get it run. Nextcloud is so a shitty software. Every new installation takes weeks of reading manuals and videos, because every time a new barrier is set up. Last Time I struggled with the certificate, now it is something unknown. May be, that also apple has its part on it.
How can I get a running system? Which Domain-supplier do you use? I spend money for my own domain but it does not work. I am so frustrated. At the time I am at Ionos, but here it does not seem to work. I don't want to use cloudflare.
Any hints or help? 
At the moment I have dyndns activated, NPM running. Nextcloud app works perfectly, only calendar and contact app won't connect. 
I tried all variations of txt and SRV settings, no success. Any hints how to debug?
Any hints for a working Domain-Provider?
Greetings from a deeply frustrated man
